napari/_vispy/vispy_image_layer.py

```diff
@@ -95,21 +86,21 @@ def _on_data_change(self, event=None):
                 self.node.set_data(data, clim=self.layer.contrast_limits)
         self.node.update()
 
-    def _on_interpolation_change(self, event=None):
+    def _on_interpolation_change(self, interpolation):
         if self.layer.dims.ndisplay == 3 and isinstance(self.layer, Labels):
             self.node.interpolation = 'nearest'
         elif self.layer.dims.ndisplay == 3 and isinstance(self.layer, Image):
             self.node.interpolation = 'linear'
         else:
-            self.node.interpolation = self.layer.interpolation
+            self.node.interpolation = interpolation
 
-    def _on_rendering_change(self, event=None):
+    def _on_rendering_change(self, value):
         if self.layer.dims.ndisplay == 3:
-            self.node.method = self.layer.rendering
-            self._on_threshold_change()
+            self.node.method = value
+            self._on_iso_threshold_change(value=None)
 
-    def _on_colormap_change(self, event=None):
-        cmap = self.layer.colormap[1]
+    def _on_colormap_change(self, value=None):
+        cmap = value if value else self.layer.colormap[1]
```

Comment:
I'm revisiting this PR now see https://github.com/shanaxel42/napari/pull/2, and agreed this is a bit strange. Curious how @napari/core-devs think we should resolve it?

It's hard for me to see how we'd only use the values passed in the handler, for example the next line requires the `layer.gamma` property even though this is the `_on_colormap_change` handler as you need to know both `gamma` and `colormap` to draw the right thing.

It's also a bit funny now because the type of the value passed to the event handler can be different now if it comes from the view side or the model side - here for example the model passes a tuple for the colormap whereas the view passes a string. Now maybe that's a symptom of a deeper problem, but it's hard for me to see how to resolve all this